### PR TITLE
Issue 2603

### LINKED ISSUE #2603: [Digit classifier] Migrate to ROS2 humble / superthin template

### PR DESCRIPTION
**Pull request by MihirGore23**
Add digit classifier newmanager

### CODE REVIEW

**Comment by javizqh** [comment]
Please, if possible use the 4.6.4 or the current latest version of robotics backend for developing and not the old robotics-academy/4.5.9

**Comment by javizqh** [comment]
Also if possible please check out the new way to create the GUI.py with this example https://github.com/JdeRobot/RoboticsAcademy/tree/humble-devel/exercises/static/exercises/follow_line_newmanager/python_template/ros2_humble

**Comment by javizqh** [comment]
Right now the frontend is not being displayed correctly:
Screenshot from 2024-07-30 11-31-12

and the RAM also gives an error when loading the world:
``` bash
developer-container  | 09:36:01 [Thread-2 (pr] [INFO ] (my_app_logger)  message received: {"id":"58d9d9e3-336b-4f77-a0b6-6636203186d5","command":"launch_world","data":{"name":"dl_digit_classifier_ros2","launch_file_path":"/opt/jderobot/Launchers/digit_classifier.launch.py","ros_version":"ROS2","visualization":"physic_rae","world":"physical","template":"RoboticsAcademy/exercises/static/exercises/dl_digit_classifier_newmanager/python_template/","exercise_id":"dl_digit_classifier_newmanager"}} from client {'id': 1, 'handler': <websocket_server.websocket_server.WebSocketHandler object at 0x73bd7c68fa30>, 'address': ('172.18.0.1', 35908)}
developer-container  | 09:36:01 [MainThread  ] [ERROR] (my_app_logger)  '2'
developer-container  | Traceback (most recent call last):
developer-container  |   File "//RoboticsAcademy/src/manager/manager/manager.py", line 507, in start
developer-container  |     self.process_message(message)
developer-container  |   File "//RoboticsAcademy/src/manager/manager/manager.py", line 408, in process_message
developer-container  |     self.trigger(message.command, data=message.data or None)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 922, in _get_trigger
developer-container  |     return event.trigger(model, *args, **kwargs)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 402, in trigger
developer-container  |     return self.machine._process(func)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1211, in _process
developer-container  |     return trigger()
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 416, in _trigger
developer-container  |     self._process(event_data)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 439, in _process
developer-container  |     if trans.execute(event_data):
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 273, in execute
developer-container  |     event_data.machine.callbacks(itertools.chain(event_data.machine.before_state_change, self.before), event_data)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1146, in callbacks
developer-container  |     self.callback(func, event_data)
developer-container  |   File "/usr/local/lib/python3.10/dist-packages/transitions/core.py", line 1165, in callback
developer-container  |     func(event_data)
developer-container  |   File "//RoboticsAcademy/src/manager/manager/manager.py", line 214, in on_launch_world
developer-container  |     self.world_launcher.run()
developer-container  |   File "/RoboticsAcademy/src/manager/manager/launcher/launcher_world.py", line 63, in run
developer-container  |     for module in worlds[self.world][str(self.ros_version)]:
developer-container  | KeyError: '2'```

**Review by javizqh — CHANGES_REQUESTED** [review]
Please do not upload the scripts files

File: scripts/develop_academy.sh
Comment (javizqh) [inline scripts/develop_academy.sh:1]: Please make sure this file is the same as the one in the humble-devel branch

File: scripts/run_academy.sh
Comment (javizqh) [inline scripts/run_academy.sh:1]: Please make sure this file is the same as the one in the humble-devel branch

**Review by javizqh — CHANGES_REQUESTED** [review]
Please change the files requested and if possible remove the files that are not from the human_detection_newmanager (all of the files from the follow line exercise)

**Comment by pawanw17** [comment]
Hi @MihirGore23, Can you please breakdown this PR to smaller ones, they are easier to review and test. also please rebase with the current humble-devel, as this PR will likely have merge issues.

**Comment by MihirGore23** [comment]
Hi.Would you please elaborate a bit further on what is expected by 'breaking down the PR to smaller ones' ?

**Comment by pawanw17** [comment]
Hi @MihirGore23, Sure. What I mean to say is divide this PR into smaller PRs, 3-4 files each or based on the parts of code they deal with... so it is easier to review them...
For eg: If you are making changes to three exercises, have a separate PR for each of the exercise

**Comment by javizqh** [comment]
Can this PR be closed?

**Comment by MihirGore23** [comment]
Yes this one can be closed

**Pull request closed without merging**